nodejs/examples/alb/ec2Instance/index.ts

```diff
@@ -16,52 +16,57 @@ import * as pulumi from "@pulumi/pulumi";
 import * as aws from "@pulumi/aws";
 import * as awsx from "@pulumi/awsx";
 
-const config = new pulumi.Config("aws");
-const providerOpts = { provider: new aws.Provider("prov", { region: <aws.Region>config.require("envRegion") }) };
+export = async () => {
```

Comment:
Does this example actually need to be async? Doesn't appear there is fundamentally any control-flow dependent on data sources.

Reply:
In this case, there is.  Specifically:

```ts
    const subnets = await vpc.publicSubnets;
    for (let i = 0; i < subnets.length; i++) {
```

This is the one example we have that actually says: go and iterate over the async data we have (the actual subnets) in order ot make new resources.